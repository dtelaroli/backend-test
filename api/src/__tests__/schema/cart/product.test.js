const { seed, mutate, query } = require("../../helpers");

const { PRODUCT, PRODUCT_UPDATE, PRODUCT_DELETE } = require("../../../db/seeders/20200923034056-product");

const {
  queries: { LIST_PRODUCTS },
  mutations: { CREATE_PRODUCT, UPDATE_PRODUCT, DELETE_PRODUCT },
} = require("../graphql");

describe("Sku", () => {
  beforeEach(() => {
    seed("20200923034056-product");
  });

  describe("Query", () => {
    describe("listProducts", () => {
      it("Should list all products", async () => {
        const result = await query({
          query: LIST_PRODUCTS,
        });

        expect(result.data.listProducts.length).toBeGreaterThan(1);
      });
    });
  });

  describe("Mutation", () => {
    const ID = "11111111-1111-1111-1111-111111111110";

    describe("createProduct", () => {
      it("Should create product", async () => {
        const result = await mutate({
          mutation: CREATE_PRODUCT,
          variables: {
            input: PRODUCT,
          },
        });

        expect(result).toMatchSnapshot();
      });

      it("Should delete product", async () => {
        const result = await mutate({
          mutation: DELETE_PRODUCT,
          variables: {
            id: PRODUCT_DELETE.id,
          },
        });

        expect(result).toMatchSnapshot();
      });

      it("Should update product", async () => {
        const result = await mutate({
          mutation: UPDATE_PRODUCT,
          variables: {
            input: {
              id: PRODUCT_UPDATE.id,
              name: "product3",
              image: "image1.jpg",
            },
          },
        });

        expect(result).toMatchSnapshot();
      });
    });
  });
});
