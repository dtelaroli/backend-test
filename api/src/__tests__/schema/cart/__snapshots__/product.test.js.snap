// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sku Mutation createProduct Should create product 1`] = `
Object {
  "data": Object {
    "createProduct": Object {
      "id": "g",
      "image": "image1.jpg",
      "name": "product1",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {
        "Cache-Control": Array [
          "max-age=5, public",
        ],
      },
    },
  },
}
`;

exports[`Sku Mutation createProduct Should delete product 1`] = `
Object {
  "data": Object {
    "deleteProduct": Object {
      "message": "Product removed",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {
        "Cache-Control": Array [
          "max-age=5, public",
        ],
      },
    },
  },
}
`;

exports[`Sku Mutation createProduct Should update product 1`] = `
Object {
  "data": Object {
    "updateProduct": Object {
      "id": "h",
      "image": "image1.jpg",
      "name": "product3",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {
        "Cache-Control": Array [
          "max-age=5, public",
        ],
      },
    },
  },
}
`;
